### PR TITLE
Improve filebeat in controller and compute nodes

### PR DESCRIPTION
**Pull request by Ruoyu111**
Modified Filebeat code. Filebeat will add 'type' tag to each service logs it collected. This would make searching and visualizing in Kibana easier.

### CODE REVIEW

**Comment by kamfonik** [comment]
This looks good, but can you align your  => characters? See the puppet style guide: https://docs.puppet.com/guides/style_guide.html#arrow-alignment

Rahul is very picky about these ;-)

**Comment by Ruoyu111** [comment]
Hi Kamfonik, thanks for the comment! I will modify it now.

File: quickstack/manifests/compute_common.pp [hunk 1]
```diff
@@ -426,18 +426,44 @@
...
-      'logstash'  => {
-        'hosts'       =>  [$elasticsearch_host],
+      'logstash' => {
+        'hosts'       => [$elasticsearch_host],
         'loadbalance' => true
       }
     },
     logging => {
-      'level' => "info"
+      'level'    => "info"
```
Comment (kamfonik) [inline quickstack/manifests/compute_common.pp:435]: I think this line and the similar line in controller_common.pp are OK without spaces actually, it is inside its own {}.  But, you can fix it next time.

**Comment by kamfonik** [comment]
+1

**Comment by Ruoyu111** [comment]
No problem. Thank you!

**Comment by rahulait** [comment]
+1. Just a thought: what will happen if a given file is not present? Like if /var/log/nova/nova.log is not present? Will it fail? I have this query because I was thinking maybe we should collect sahara logs as well. In that case, we might enable sahara on some and disable on others, so will that affect this?

**Comment by rahulait** [comment]
Merging this one for now.

**Comment by Ruoyu111** [comment]
Hi Rahul, thanks for the merging and comment! Nothing will happen if Filebeat was configured to collect a log file but that log file doesn't exist. If Filebeat cannot find the log file in the given path, it will continue to find other log files in its list. And filebeat will continue checking whether there are new contents in log files on its list. If there are new log contents, it will collect them and transfer them.
It should be fine to let filebeat collect sahara logs on all nodes even if some nodes don't have sahara installed.

**Pull request merged**